test/unit/p2p_blocks_test.exs

```diff
@@ -0,0 +1,39 @@
+defmodule Unit.P2PBlocks do
+  use ExUnit.Case, async: true
+  use Patch
+
+  alias LambdaEthereumConsensus.Store.BlockStore
+  alias LambdaEthereumConsensus.P2P.BlockDownloader
+  alias LambdaEthereumConsensus.Libp2pPort
+  alias LambdaEthereumConsensus.P2P.IncomingRequests.Handler
+  alias SszTypes.BeaconBlocksByRootRequest
+
+  setup do
+    expose(Handler, :all)
+  end
+
+  test "one block retrieve" do
+    #
+    signed_block_input = Fixtures.Block.signed_beacon_block()
+    block_root = Ssz.hash_tree_root!(signed_block.message)
+
+    # store the block in the BlockStore
+    BlockStore.store_block(signed_block_input)
+
+    # ssz serialize and snappy compress the block root
+    with {:ok, ssz_serialized} <- Ssz.to_ssz(%BeaconBlocksByRootRequest{body: [block_root]}),
+         {:ok, snappy_compressed_message} <- Snappy.compress(ssz_serialized) do
+      # patch the Libp2pPort's send_request function to call the incoming_requests handler function we want to test
+      patch(Libp2pPort, :send_request, fn _peer_id, _protocol, _message ->
+        Handler.handle_req("beacon_blocks_by_root/2/ssz_snappy", 1, snappy_compressed_message)
```

Comment:
We have to pass the actual message instead of building it from scratch.
```suggestion
      patch(Libp2pPort, :send_request, fn _peer_id, _protocol, message ->
        Handler.handle_req("beacon_blocks_by_root/2/ssz_snappy", 1, message)
```